### PR TITLE
Add io.qt.PySide.BaseApp

### PR DESCRIPTION
**Pull request by shyamnathp**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.
Flatpak base app for PySide6, can be used as the base app for packaging Flatpak applications that use PySide6. [PySide6](https://doc.qt.io/qtforpython-6/) offers the official Python bindings for [Qt](https://doc.qt.io/), which enables you to use Python to write your Qt applications

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:** https://wiki.qt.io/Maintainers#Qt_for_Python_Maintainers

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements#control-over-domain
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.qt.PySide.BaseApp.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="runtime">
+    <id>io.qt.PySide6.BaseApp</id>
+    <name>PySide6 BaseApp</name>
+    <metadata_license>CC0-1.0</metadata_license>
+    <project_license>LicenseRef-free=https://doc.qt.io/qtforpython-6/</project_license>
```
Comment (bbhtt) [inline io.qt.PySide.BaseApp.metainfo.xml:6]: This should be LGPL right? The commercial part is provided as addons afaik and is not being built here?

Comment (shyamnathp) [reply io.qt.PySide.BaseApp.metainfo.xml]: right. The validation with flatpak-builder-lint showed me a warning to use LicenseRef. I have updated it.

Comment (bbhtt) [reply io.qt.PySide.BaseApp.metainfo.xml]: That's fine, info hints are not errors.

**Comment by bbhtt** [comment]
You have to add patchelf.

Also was this locally built using Flatpak? I never managed to build it using python setup.py in a Flatpak environment due to some issues with prefixes I don't remember.

The way I build it from source is using cmake/ninja https://github.com/flathub/io.github.hydrusnetwork.hydrus/blob/72035d3869bff2e56eb62a69340599183cf2927c/io.github.hydrusnetwork.hydrus.yml#L273-L276

**Comment by bbhtt** [comment]
The appstream ID tag and Flatpak ID does not match. One has Pyside6, one does not. That's the filters-but-no-output error.

**Comment by shyamnathp** [comment]
> You have to add patchelf.
> 
> Also was this locally built using Flatpak? I never managed to build it using python setup.py in a Flatpak environment due to some issues with prefixes. I don't exactly remember what the issues were.
> 
> The way I build it from source is building shiboken6 first then using cmake/ninja for the rest https://github.com/flathub/io.github.hydrusnetwork.hydrus/blob/72035d3869bff2e56eb62a69340599183cf2927c/io.github.hydrusnetwork.hydrus.yml#L273-L276

Hey :), 

Yes, this was tested locally and even build another Flatpak application with this baseapp. That seemed to work well as well. 

The "python setup.py" for Qt for Python build, basically builds and calls a series of 3 CMake commands for the targets shiboken6, shiboken6_generator and PySide6. This can be seen if you see the build that succeeded on x86_64. It would have the following line:

`Running command:  /usr/bin/cmake -DCMAKE_UNITY_BUILD=ON -DCMAKE_UNITY_BUILD_BATCH_SIZE=16 ....`

 The advantage of "setup.py" build here is that it even does the copying of the binaries for all these 3 targets into "site-packages" directory of the concerned Python, which for a CMake/Ninja case I will have to do manually. 

One of the issues where the "setup.py" command was earlier not working is because for the Kde Flatpak runtime, the Qt headers were system headers and libclang (used by shiboken6_generator) ignores it. This is fixed by the attached patch 'shiboken_force_process_system_headers.patch'

I am not sure why patchelf installation failed for aarch64. Investigating.

**Comment by shyamnathp** [comment]
The "patchelf" missing issue was because flatpak-pip-generator only considered x86_64 wheels when i created it. Hopefully, it should be fixed now.

**Comment by bbhtt** [comment]
bot, build io.qt.PySide.BaseApp

**Comment by bbhtt** [comment]
Seems to work.

**Comment by shyamnathp** [comment]
> Any idea why the pdf modules are skipped? Afaik Qtwebengine should be enough to have it.
> 
> ```
> -- optional module Qt6Pdf skipped. Looked in: /app/lib/x86_64-linux-gnu/cmake/Qt6Pdf
> -- optional module Qt6PdfWidgets skipped. Looked in: /app/lib/x86_64-linux-gnu/cmake/Qt6PdfWidgets
> ```
> 
> I noticed this issue earlier too in [flathub/io.github.hydrusnetwork.hydrus#31](https://github.com/flathub/io.github.hydrusnetwork.hydrus/issues/31) and now here again but never found a fix.

Interesting. As i check the 'org.kde.Sdk//6.7', the headers for QtPdf and QtPdfWidgets are missing in '/usr/include'. I am not sure why Kde excludes them, but since shiboken cannot find the headers, it cannot create the Python bindings for it.

**Comment by bbhtt** [comment]
They should be in the Qtwebengine baseapp right?

**Comment by bbhtt** [comment]
Yea

```
flatpak run --command=ls io.qt.qtwebengine.BaseApp//6.7 -lA "/app/include/QtPdf"
                                                                            15s
total 96
drwxr-xr-x. 1 bbhtt bbhtt   10 Jan  1  1970 6.7.2
-rw-r--r--. 2 bbhtt bbhtt   54 Jan  1  1970 QPdfBookmarkModel
-rw-r--r--. 2 bbhtt bbhtt 1616 Jan  1  1970 qpdfbookmarkmodel.h
-rw-r--r--. 2 bbhtt bbhtt   49 Jan  1  1970 QPdfDocument
-rw-r--r--. 2 bbhtt bbhtt 3234 Jan  1  1970 qpdfdocument.h
-rw-r--r--. 2 bbhtt bbhtt   62 Jan  1  1970 QPdfDocumentRenderOptions
-rw-r--r--. 2 bbhtt bbhtt 2847 Jan  1  1970 qpdfdocumentrenderoptions.h
-rw-r--r--. 2 bbhtt bbhtt   45 Jan  1  1970 QPdfLink
-rw-r--r--. 2 bbhtt bbhtt 2376 Jan  1  1970 qpdflink.h
-rw-r--r--. 2 bbhtt bbhtt   50 Jan  1  1970 QPdfLinkModel
-rw-r--r--. 2 bbhtt bbhtt 1568 Jan  1  1970 qpdflinkmodel.h
-rw-r--r--. 2 bbhtt bbhtt   54 Jan  1  1970 QPdfPageNavigator
-rw-r--r--. 2 bbhtt bbhtt 1815 Jan  1  1970 qpdfpagenavigator.h
-rw-r--r--. 2 bbhtt bbhtt   53 Jan  1  1970 QPdfPageRenderer
-rw-r--r--. 2 bbhtt bbhtt 1751 Jan  1  1970 qpdfpagerenderer.h
-rw-r--r--. 2 bbhtt bbhtt   52 Jan  1  1970 QPdfSearchModel
-rw-r--r--. 2 bbhtt bbhtt 1754 Jan  1  1970 qpdfsearchmodel.h
-rw-r--r--. 2 bbhtt bbhtt   50 Jan  1  1970 QPdfSelection
-rw-r--r--. 2 bbhtt bbhtt 1903 Jan  1  1970 qpdfselection.h
-rw-r--r--. 2 bbhtt bbhtt  402 Jan  1  1970 QtPdf
-rw-r--r--. 1 bbhtt bbhtt    0 Jul 22 20:44 qtpdf-config.h
-rw-r--r--. 2 bbhtt bbhtt  181 Jan  1  1970 QtPdfDepends
-rw-r--r--. 2 bbhtt bbhtt 1777 Jan  1  1970 qtpdfexports.h
-rw-r--r--. 2 bbhtt bbhtt  276 Jan  1  1970 qtpdfglobal.h
-rw-r--r--. 2 bbhtt bbhtt   49 Jan  1  1970 QtPdfVersion
-rw-r--r--. 2 bbhtt bbhtt  192 Jan  1  1970 qtpdfversion.h
```

**Comment by hfiguiere** [comment]
In the `/app` prefix which is part of the problem with pyside as it doesn't support prefix. And require patching. It has been hard to maintain the FreeCAD flatpak due to these bugs.

**Comment by shyamnathp** [comment]
ah i see now. The thing is the build uses 'qtpaths' or 'qmake' to find the path to the headers. Since, I pass /usr/bin/qtpaths it only find the headers from '/usr/include' from the sdk which does not include Qt modules like QtPdf, QtWebEngine etc.

**Comment by bbhtt** [comment]
Is it possible to pass extra include paths somehow?

**Comment by shyamnathp** [comment]
I was just contemplating on the idea. Currently, it is not possible. I can create a patch in the coming days to PySide6 to do that, and then add the patch along with the flatpak recipe.

**Comment by bbhtt** [comment]
Ok sure if you can get it working with the split prefixes that'd be great.

**Comment by shyamnathp** [comment]
Thanks for merging this in the time being. I will fix the prefix situation soon.

**Comment by shyamnathp** [comment]
The prefix situation should be fixed with https://github.com/pyside/io.qt.PySide.BaseApp/pull/1 and the missing modules should be available after it is merged. 

The fork within the PySide organization is basically to control all the merges to flathub/io.qt.PySide.BaseApp, and also to have proper review from everyone involved in the Qt for Python project. 

Can someone please help me to make the [wiki](https://github.com/flathub/io.qt.PySide.BaseApp/tree/wiki) branch as the default branch for flathub/io.qt.PySide.BaseApp. Thanks a ton.

**Comment by bbhtt** [comment]
Changed, you can open an issue for these https://github.com/flathub/flathub/issues or if you need additional people added as maintainers to the repo.

**Pull request closed without merging**